nixos/modules/programs/gnupg.nix

```diff
@@ -6,6 +6,19 @@ let
 
   cfg = config.programs.gnupg;
 
+  xserverCfg = config.services.xserver;
+
+  defaultPinentryFlavour =
+    if xserverCfg.desktopManager.gnome3.enable then
```

Comment:
Could you add pantheon?

Reply:
done, also updated the description to include pantheon.